assets/src/edit-story/components/carousel/carouselProvider.js

```diff
@@ -44,6 +44,12 @@ function CarouselProvider({ availableSpace, children }) {
 
   const pageIds = useMemo(() => pages.map(({ id }) => id), [pages]);
 
+  const [showSkeleton, setShowSkeleton] = useState(true);
+
+  useEffect(() => {
+    requestIdleCallback(() => setShowSkeleton(false), { timeout: 5000 });
```

Comment:
Does this need a `cancelIdleCallback` on unmount?

Reply:
CMIIW: even if the user keeps refreshing the page before this callback fires (at max 5s) it will not cause memory leaks because the whole window is either way destroyed 🤔 . Of course, it can be done for good practice.

Reply:
Yeah, the component only appears once and is re-rendered every now and then, but never fully unmounts. But that might change in the future and it's good practice, so will add it in.